tools/c7n_logexporter/c7n_logexporter/exporter.py

```diff
@@ -189,36 +190,65 @@ def converge_destination_policy(client, config):
-        session = get_session(t_account['role'])
+    def subscribe_account(t_account, subscription, region):
+        session = get_session(t_account['role'], region)
         client = session.client('logs')
-
+        sub_name = subscription.get('name', 'FlowLogStream')
+        distribution = subscription.get('distribution', 'ByLogStream')
+        
         for g in account.get('groups'):
-            sub_name = subscription.get('name', 'FlowLogStream')
-            distribution = subscription.get('distribution', 'ByLogStream')
-            filters = client.describe_subscription_filters(
-                logGroupName=g,
-                ).get('subscriptionFilters', ())
-            if filters:
-                f = filters.pop()
-                if (f['filterName'] == sub_name
-                    and f['destinationArn'] == subscription['destination-arn']
-                    and f['roleArn'] == subscription['destination-role']
-                    and f['distribution'] == distribution):
-                    continue
-                client.delete_subscription_filter(
-                    logGroupName=g, filterName=sub_name)
-            client.put_subscription_filter(
-                logGroupName=g,
-                roleArn=subscription['destination-arn'],
-                destinationArn=subscription['destination-arn'],
-                filterName=sub_name,
-                filterPattern="",
-                distribution=distribution)
+            if (g.endswith('*')):
+                g = g.replace('*','')
+                allLogGroups = []
+                paginator = client.get_paginator('describe_log_groups')
+                for p in paginator.paginate(logGroupNamePrefix=g):
+                    loggroupz = p['logGroups']
+                    for logz in loggroupz:
+                        allLogGroups.append(logz['logGroupName'])
+                
+                for l in allLogGroups:
+                    filters = client.describe_subscription_filters(
+                        logGroupName=l,
+                        ).get('subscriptionFilters', ())
+                    if filters:
+                        f = filters.pop()
+                        if (f['filterName'] == sub_name
+                            and f['destinationArn'] == subscription['destination-arn']
+                            and f['distribution'] == distribution):
+                            continue
+                        client.delete_subscription_filter(
+                            logGroupName=l, filterName=sub_name)
+                    client.put_subscription_filter(
+                        logGroupName=l,
+                        destinationArn=subscription['destination-arn'],
+                        filterName=sub_name,
+                        filterPattern="",
+                        distribution=distribution)
+            else:
+                sub_name = subscription.get('name', 'FlowLogStream')
```

Comment:
looks like this logic is being repeated, would be preferable to move this into a separate function to prevent violation of DRY

Reply:
@thisisshi  Which logic is it that is concerning?  and what is a DRY violation?  Sorry I am pretty new to Python so please excuse my dumb questions.   and thanks for your help!

Reply:
I'm on it.

Reply:
@jtroberts83 no problem, DRY is just an acronym for "Don't Repeat Yourself", it's better to separate repeated code into its own function so it's easier to handle in the future :)